test_utils/src/lib.rs

```diff
@@ -1 +1,96 @@
+//! Utilities for use in tests.
+
 pub mod tx_data;
+
+use std::env;
+use std::path::PathBuf;
+
+use strum::EnumIter;
+
+/// Path from the root of the Git repo to the directory under which built test
+/// wasms can be found.
+pub const WASM_FOR_TESTS_DIR: &str = "wasm_for_tests";
+
+/// Corresponds to wasms that we build for tests, under [`WASM_FOR_TESTS_DIR`].
+/// See the `wasm_for_tests/wasm_source` crate for documentation on what these
+/// wasms do.
+#[allow(missing_docs)]
+#[derive(Debug, Clone, Copy, EnumIter)]
+pub enum TestWasms {
+    TxMemoryLimit,
+    TxMintTokens,
+    TxNoOp,
+    TxProposalCode,
+    TxReadStorageKey,
+    TxWriteStorageKey,
+    VpAlwaysFalse,
+    VpAlwaysTrue,
+    VpEval,
+    VpMemoryLimit,
+    VpReadStorageKey,
+}
+
+impl TestWasms {
+    /// Get the path to where this test wasm is expected to be, or panic if not
+    /// able to.
+    pub fn path(&self) -> PathBuf {
+        let filename = match self {
+            TestWasms::TxMemoryLimit => "tx_memory_limit.wasm",
+            TestWasms::TxMintTokens => "tx_mint_tokens.wasm",
+            TestWasms::TxNoOp => "tx_no_op.wasm",
+            TestWasms::TxProposalCode => "tx_proposal_code.wasm",
+            TestWasms::TxReadStorageKey => "tx_read_storage_key.wasm",
+            TestWasms::TxWriteStorageKey => "tx_write.wasm",
+            TestWasms::VpAlwaysFalse => "vp_always_false.wasm",
+            TestWasms::VpAlwaysTrue => "vp_always_true.wasm",
+            TestWasms::VpEval => "vp_eval.wasm",
+            TestWasms::VpMemoryLimit => "vp_memory_limit.wasm",
+            TestWasms::VpReadStorageKey => "vp_read_storage_key.wasm",
+        };
+        let cwd =
+            env::current_dir().expect("Couldn't get current working directory");
+        // crudely find the root of the repo, we can't rely on the `.git`
+        // directory being present, so look instead for the presence of a
+        // CHANGELOG.md file
+        let repo_root = cwd
+            .ancestors()
+            .find(|path| path.join("CHANGELOG.md").exists())
+            .unwrap_or_else(|| {
+                panic!(
+                    "Couldn't find the root of the repository for the current \
+                     working directory {}",
+                    cwd.to_string_lossy()
+                )
+            });
```

Comment:
We don't have the `.git` directory when running tests in CI, so this code crudely looks for the ancestory directory which has a `CHANGELOG.md` file, there is probably a better way to do this.